connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -912,6 +912,11 @@ object SparkSession extends Logging {
     @deprecated(message = "Please use create() instead.", since = "3.5.0")
     def build(): SparkSession = create()
 
+    private def getClientConfiguration(): Configuration = {
+      // Use copy() to avoid modifying the state of the existing builder.
+      builder.copy().loadFromEnvironment().configuration
```

Comment:
in this case, SPARK_REMOTE will take precedence over what was explicitly set in the builder, right?
so having in env
```
SPARK_REMOTE="sc://foo:123"
```
and calling
```
SparkSession.builder.remote("sc://bar:234").create()
```
then the SPARK_REMOTE will take precedence? Documentation does not specify precedence, but it feels like setting `remote` explicitly should take precedence over env var (and it should also be specified in the documentation)

Reply:
In python, the explicity set remote config set in https://github.com/apache/spark/blob/master/python/pyspark/sql/connect/session.py#L143 takes precedence over env in https://github.com/apache/spark/blob/master/python/pyspark/sql/session.py#L470

It should be the same in scala, and the precedence should be documented.

Reply:
SPARK_REMOTE value makes the final override before [RemoteSparkSession.builder.config(map=opts).getOrCreate()](https://github.com/apache/spark/blob/master/python/pyspark/sql/session.py#L479) if I understand correctly?